packages/stream/components/Suspense.astro

```diff
@@ -1,8 +1,49 @@
 ---
-const idx = Astro.locals.suspend(Astro.slots.render("default"));
----
+import { trackThenableState, sleep } from "./utils";
+
+const thenable = trackThenableState(Astro.slots.render("default"));
+
+// Wait a moment to see if the slot renders quickly enough --
+// if it does, there's no need to send a fallback.
+const DEADLINE_MS = 5;
+await sleep(DEADLINE_MS);
 
-<div style="display: contents" data-suspense-fallback={idx}>
-  <slot name="fallback" />
-</div>
+type RenderKind = { kind: 'fallback', id: number } | { kind: 'content', content: string }
```

Comment:
@bholmesdev `getThenableState` ended up not helping much, because i still needed to store the boundary id or content. settled on something like this, what do you think?

Reply:
I like this a lot! 👍